File: 485.Max-Consecutive-Ones.py
# https://leetcode.com/problems/max-consecutive-ones/
#
# algorithms
# Easy (54.52%)
# Total Accepted:    124,735
# Total Submissions: 228,773
# beats 100.0% of python submissions


class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        cur_ones = 0

        for n in nums:
            if n == 1:
                cur_ones += 1
            else:
                res = max(res, cur_ones)
                cur_ones = 0

        return max(res, cur_ones)

File: 485.Max-Consecutive-Ones.java
// https://leetcode.com/problems/max-consecutive-ones/
//
// algorithms
// Easy (55.76%)
// Total Accepted: 158,915
// Total Submissions: 284,984


class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0;
        int tmp = 0;

        for (int n : nums) {
            if (n == 0) {
                res = Math.max(res, tmp);
                tmp = 0;
            } else {
                tmp++;
            }
        }

        return Math.max(res, tmp);
    }
}